### PR TITLE
Topic1

### PR DESCRIPTION
**Pull request by facusuarez**
Finished Topic 1.

### CODE REVIEW

File: facundo/01-JS-DOM-APIs/Exercise10/index.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+<head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <link href="https://necolas.github.io/normalize.css/5.0.0/normalize.css" rel="stylesheet">
+    <link href="style.css" rel="stylesheet">
+    <title>JavaScript</title>
+    <script type="text/javascript" src="script.js"></script>
+</head>
+
+<body onload="fadein()">
```
Comment (marcelososa) [inline facundo/01-JS-DOM-APIs/Exercise10/index.html]: Remember, we should to write code in unobtrusive way, any listener should added from our js code, not in the HTML.

File: facundo/01-JS-DOM-APIs/Exercise10/index.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    <header>
+        <h1>I'm a header</h1>
+    </header>
+    <div id="lista"><div id="titu">Repositories List</div></div>
+    <section class="hidden" id="1">
```
Comment (marcelososa) [inline facundo/01-JS-DOM-APIs/Exercise10/index.html]: Use a representative ID for elements.

File: facundo/01-JS-DOM-APIs/Exercise10/script.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            });
+        };
+        xhr.send();
+    });
+}
+function loadJoke() {
+    var obj = { method: "GET", url: "http://api.icndb.com/jokes/random", str: null };
+    makeRequest(obj)
+        .then(function (value) {
+            content = JSON.parse(value)
```
Comment (marcelososa) [inline facundo/01-JS-DOM-APIs/Exercise10/script.js]: This variable will be declared globally if you don't define its scope.

File: facundo/01-JS-DOM-APIs/Exercise10/script.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        })
+        .catch(function (err) {
+            console.error('Error!', err.statusText);
+        });
+}
+function loadRepository() {
+    var obj = { method: "GET", url: "https://api.github.com/search/repositories?q=" + document.getElementById("q").value };
+    makeRequest(obj)
+        .then(function (value) {
+            content = JSON.parse(value)
```
Comment (marcelososa) [inline facundo/01-JS-DOM-APIs/Exercise10/script.js]: This variable will be declared globally if you don't define its scope.

File: facundo/01-JS-DOM-APIs/exercise12/script.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+function createArray(fil, col) {
+    var array = [];
+    for (var i = 0; i < fil; i++) {
+        array[i] = [];
+        for (var j = 0; j < col; j++) {
+            array[i][j] = (Math.random() * 100 | 0);
+        }
+    }
+    createTable(array)
```
Comment (marcelososa) [inline facundo/01-JS-DOM-APIs/exercise12/script.js]: I think that this function should return an array and not call another function, isn't its responsability to create a table.

File: facundo/01-JS-DOM-APIs/Exercise10/index.html (hunk 1 above)
Comment (facusuarez) [inline facundo/01-JS-DOM-APIs/Exercise10/index.html]: Fixed

File: facundo/01-JS-DOM-APIs/Exercise10/index.html (hunk 2 above)
Comment (facusuarez) [inline facundo/01-JS-DOM-APIs/Exercise10/index.html]: Fixed

File: facundo/01-JS-DOM-APIs/Exercise10/script.js (hunk 3 above)
Comment (facusuarez) [inline facundo/01-JS-DOM-APIs/Exercise10/script.js]: Fixed

File: facundo/01-JS-DOM-APIs/Exercise10/script.js (hunk 4 above)
Comment (facusuarez) [inline facundo/01-JS-DOM-APIs/Exercise10/script.js]: Fixed

File: facundo/01-JS-DOM-APIs/Exercise10/script.js (hunk 3 above)
Comment (marcelososa) [inline facundo/01-JS-DOM-APIs/Exercise10/script.js]: I don't see changes here.

File: facundo/01-JS-DOM-APIs/Exercise10/script.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        })
+        .catch(function (err) {
+            console.error('Error!', err.statusText);
+        });
+}
+function loadRepository() {
+    let obj = { method: "GET", url: "https://api.github.com/search/repositories?q=" + document.getElementById("q").value };
+    makeRequest(obj)
+        .then(function (value) {
+            content = JSON.parse(value)
```
Comment (marcelososa) [inline facundo/01-JS-DOM-APIs/Exercise10/script.js]: Here neither

Comment (facusuarez) [inline facundo/01-JS-DOM-APIs/Exercise10/script.js]: Oops, Don't know where the changes went. It's Fixed now. But i have to ask, is that a good solution? I thought the `var` was scoped in the function... What would have been your solution?

Comment (marcelososa) [inline facundo/01-JS-DOM-APIs/Exercise10/script.js]: Hi Facu, I refer to the `content` variable, if you don't declare into the `loadJoke` as `var`, `let` or `const` its be declared and asigned value in the same step but in the global context.

Change your function to:

```javascript
function loadJoke() {
    makeRequest({ method: "GET", url: "http://api.icndb.com/jokes/random", str: null })
        .then(function (value) {
            var content = JSON.parse(value)
            document.getElementById("HelloWorld").innerHTML = content.value.joke
        })
        .catch(function (err) {
            console.error('Error!', err.statusText);
        });
}
```
the same in `loadRepository` function.

Another exmaple (copy and paste in the browser console):

```javascript
console.log(_temp); // Reference error
function test() {
    _temp = 'some'; // Declare and assign value, automatically in the global context
}
console.log(_temp); // Output: "some"
```
fixed
```javascript
console.log(_temp); // Reference error
function test() {
    var _temp = 'some'; // Declare and assign value in the current context
}
console.log(_temp); // Reference error
```

Comment (facusuarez) [inline facundo/01-JS-DOM-APIs/Exercise10/script.js]: Ahhh. Sorry, i thought the comment was about `var obj`.
 Yes, you were right, I just didn't see that bad declaration. My mistake.

**Pull request merged**